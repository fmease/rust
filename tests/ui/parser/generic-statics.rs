static S<T>: i32 = 0;
//~^ ERROR static items may not have generic parameters

static T: () = ()
where
    Vec<()>: Clone;
//~^^ ERROR static items may not have a where clause

static U: bool
where
    (): Copy,
= true;
//~^^^ ERROR static items may not have a where clause

fn main() {}
