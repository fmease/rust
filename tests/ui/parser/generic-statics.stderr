error: static items may not have generic parameters
  --> $DIR/generic-statics.rs:1:9
   |
LL | static S<T>: i32 = 0;
   |         ^^^

error: static items may not have a where clause
  --> $DIR/generic-statics.rs:5:1
   |
LL | / where
LL | |     Vec<()>: Clone;
   | |__________________^

error: static items may not have a where clause
  --> $DIR/generic-statics.rs:10:1
   |
LL | / where
LL | |     (): Copy,
   | |_____________^

error: aborting due to 3 previous errors

